torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -472,8 +484,13 @@ def local_state_dict(self) -> Dict:
         """
         return self.optim.state_dict()
 
-    def state_dict(self) -> Dict[str, Any]:
+    def state_dict(self, all_ranks: bool = False) -> Dict[str, Any]:
```

Comment:
genuine question: is that ok ? changes the interface that everybody knows, but the default `state_dict()` is still usable